### DISCUSSION TITLE
[Feature Request] Android app

### DISCUSSION THREAD

**Discussion by ados8** [donetick/donetick#51] [Ideas]
I know this is a big ask but would it be on the roadmap.
The main reason is notifications, although if support is expanded to webhooks and other apps that can be mitigated.
The second is widgets which generally is a must for tasks to see and manage them quickly.

**Comment by meauxt** [comment]
Say no more. here you go  [release ](https://github.com/donetick/donetick/releases/tag/v0.0.24) 

@ados8 please keep in mind this is alpha / a pre-release!

**Reply by Slurms92** [reply]
Any chance an iOS app is in the works?

**Reply by meauxt** [reply]
Hi @Slurms92 !Yes but for now, i shift my focus to solidifying the core functionality. I did start investigating the app store release process, but I decided to prioritize development for the time being.

While it's technically possible to build the app from the frontend repo code (with an Apple developer account), I wouldn't recommend it just yet. It's still in early beta and needs more work.

Once the core features are more polished, I'll shift my attention to the Mobile app polishing(at least the plan) releasing on the app stores.

**Reply by Slurms92** [reply]
Awesome mate. Genuinely, thank you for your work. I love donetick. I've been looking for something like this for so long and has become up there for one of my favorite self hosted apps. I can't wait to see what the future holds as it's already had a very positive impact on my life.

**Reply by janstadt** [reply]
I just posted a suggestion for a progressive webapp to get notifications and whatnot. Would love to see either that or an iOS app. Very promising start on this app btw.

**Comment by ados8** [comment]
@meauxt I've just installed, thank you 🎉.
I've started moving away from Android Tasks (org.tasks) and to Super Productivity.
I was chasing NLP (natural language processing) and while they don't have it they do have automatic tag words.
However, their app is designed around a single user not a family so they are effectively out of consideration.
Donetick having multi user accounts (although no admin management) and server side sync is a big plus over other options.

Donetick is close to perfect and I think a few more features and it would have no competition, especially if NLP was implemented.
For those not aware of what NLP does, it simplifies the task creation process.
For most users the biggest hassle with using a task app is making them and find it hard to form the habit and make it stick.
An app without NLP results in the following to make a task.

1. Select new task
2. Title the task
3. Use a date and time picker for due and/or reminder time

An app with NLP results in the following to make a task.

1. Select new task
2. Type everything in the task title i.e. Take out rubbish Friday at 6 AM

The task app interprets what the user is intending making a task named _Take out rubbish_ and sets the due date as Friday 6 AM and a reminder at that time.
Some apps also allow adding a remind 1h to instruct the app to instead set a reminder 1 hour before not at the time of task.
Currently the only task apps doing NLP are paid options and while I understand costs exist the concern for self hosting advocates is privacy.
A task app has _a lot_ of private data about you and a self hosted option is ideal, however none exist with NLP.

**Reply by meauxt** [reply]
@ados8 the [v0.0.27](https://github.com/donetick/donetick/releases/tag/v0.0.27) should have something similar but very experimental at this stage

**Reply by ados8** [reply]
That is amazing, thank you and the rest of the team for the hard work developing this out.
I might be too tired/dumb at the moment but it hasn't worked for me, just makes a task with that title.
Will revisit soon, probably all on me.

**Reply by ados8** [reply]
I've tested this on mobile and web and I've been unable to get NPL like functions.
The result is a task titled "Take out rubbish Friday at 6 AM".

**Reply by meauxt** [reply]
Hi there @ados8 !
![Screenshot_20250207-234411](https://github.com/user-attachments/assets/2659137e-711f-40e1-9994-919b57a5453d)
Are you using this text field to create the task? The task in sentence future only available there and not from the details edit view as of now

**Reply by ados8** [reply]
My fault, I was defaulting to using the task button, I don't recall there being instructions 😅.
Tested and it works, can't wait to see these features expand.
Thank you for all the hard work making this app grow in a market where there's such little competition.

**Comment by mschwerz** [comment]
@meauxt is there a repo for the Android app? I would be curious about contributing to it

**Reply by meauxt** [reply]
Hi @mschwerz ! it's the same as the frontend repo. right now using Capacitor

**Comment by ados8** [comment]
Friendly follow up on how the Android app might be coming along?
I've been installing any new updated APK but haven't been able to use it.
Continuing to use mobile browser and webhooks to another app for notifications.
Some of the issues I've run into are.

Every app open it shows nothing scheduled.
To get it to display anything I need to begin making a task and then cancel, this forces a refresh and it loads the content.
You could use another method of refresh e.g. the menu bar and then switch back but unfortunately that doesn't load either.
A similar trick to get the menu bar is to begin a new task and in the top left the hamburger menu is available, select it and then home.
This resolves the task loading issue as well, as long as you don't close the app from the app draw or you'll need to repeat the process.
This seems to be a landing page URL bug, I've attempted to force a URL redirect at web server level but hasn't got around it.

The other little issue is NFC, it wants to write using the local IP not the public DNS.
I'm using the public URL in the server setting and if I use the mobile web browser the NFC URL is the public address.

Let me know if this needs to be a GitHub issue.
Look forward to seeing widgets to reduce the need to open the app.

**Reply by ados8** [reply]
Forgot to mention that NFC doesn't work with the mobile app as it must open to a URL.
This was annoying as I had the "app" added a web app/shortcut on the home screen.
This worked for the app functions but when scanning NFC tag it would only use the browser apps and open a new tab in my many tabs.
I've found solutions to all of these issues and I'm rolling with it for now 😊.
I have a "app like" experience and NFC works too, for those wanting to copy this is what you'll need.
Note that some of these apps cost and therefore this solution might not be for you.

1. Install Tasker for Android
2. Install Hermit for Android
3. Add a new lite app from within Hermit to your Donetick web portal and save to home screen
4. Copy the URL from the NFC
5. In Tasker setup a profile for an NFC detection and scan your tag, you only need to match the ID (shorter method)
6. Make a new task in Tasker for your profile and have open the copied URL with Hermit.
7. Save the profile and ensure NFC tags open Tasker by default.

You can rewrite the tag with NFC format that Tasker better understands so you can pull the data dynamically but then other people not using this setup might have issues.
